snap/export_test.go

```diff
@@ -19,7 +19,20 @@
 
 package snap
 
+import (
+	"regexp"
+)
+
 var (
 	ImplicitSlotsForTests        = implicitSlots
 	ImplicitClassicSlotsForTests = implicitClassicSlots
+	oldSupportedHooks            = supportedHooks
 )
+
+func FakeSupportedHookType(pattern *regexp.Regexp) {
```

Comment:
The pattern we use for this sort of thing looks like this:
```
func MockSupportedHookType(...) (restore func()) {
    old := supportedHooks
    supportedHooks = pattern
    return func() { supportedHooks = old }
}
```
and in the test:
```
restore := pkg.MockFoo(...)
defer restore()
```

That said, do we need to mock that?  Might we rely on a registered hook type and test with that one instead?